examples/llama.py

```diff
@@ -340,6 +346,21 @@ def greedy_until(self, prompt:str, until, max_length, temperature):
         if output.endswith(s): return output[0:-len(s)]
     return output
 
+  @staticmethod
+  def _prepare_model(model_path, tokenizer_path, model_url, tokenizer_url):
+    os.makedirs(model_path, exist_ok=True)
+    model_safetensors_path = Path.joinpath(model_path, "model.safetensors")
+    if not model_safetensors_path.exists():
+      LLaMa._fetch_and_replace_file(model_url, model_safetensors_path)
+    if not tokenizer_path.exists():
+      LLaMa._fetch_and_replace_file(tokenizer_url, tokenizer_path)
+    return model_safetensors_path
+
+  @staticmethod
+  def _fetch_and_replace_file(url, destination_path):
+    downloaded_path = fetch(url)
+    os.replace(downloaded_path, destination_path)
```

Comment:
fetch already returns pathlib.Path so you should be able to use replace/rename from pathlib directly without calling os to do the same